src/data/list/cycle.lean

```diff
@@ -663,4 +684,7 @@ by { rw [←next_reverse_eq_prev, ←mem_reverse_iff], exact next_mem _ _ _ _ }
 
 end decidable
 
+instance [has_repr α] : has_repr (cycle α) :=
+⟨λ s, "c[" ++ string.intercalate ", " ((s.map repr).lists.sort (≤)).head ++ "]"⟩
```

Comment:
Can you add a test for this, and a docstring that gives an example of what the repr looks like?

Reply:
@eric-wieser :ping_pong:

Reply:
It's not quite as easy as I thought, I've asked at https://leanprover.zulipchat.com/#narrow/stream/239415-metaprogramming-.2F.20tactics/topic/Testing.20has_repr/near/249454872

Reply:
Ah, this seems to work:
```
open tactic

run_cmd
do x ← to_expr ``(repr (↑[2, 1, 4, 3] : cycle ℕ)),
   "c[1, 4, 3, 2]" ← eval_expr string x,
   pure ()
```

That would just go in a new file in the `test` directory.

Reply:
Ah, @ChrisHughes24 pointed out I'm majorly overcomplicating this, and the test just needs to be
```lean
run_cmd do guard ("c[1, 4, 3, 2]" = repr (↑[2, 1, 4, 3] : cycle ℕ))
```